server/src/main/java/org/opensearch/tasks/SearchBackpressureTask.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+/**
+ * A task related to search backpressure
+ *
+ * @opensearch.internal
+ */
+public interface SearchBackpressureTask {}
```

Comment:
This interface isn't adding any common functionality. Can it be removed?

Reply:
@ketanv3 this is a marker interface: the idea is to distinguish search tasks w/o backpressure support (instead of hard-coding the concrete classes as it was before).

Reply:
@reta Thanks for the clarification!